src/main/java/Playfield.java

```diff
@@ -343,6 +345,28 @@ private void place() {
         displayGrid();
     }
 
+    private int CheckClearLine() {
+        int line_cleared = 0; // Stores how many line the block cleared
+        // Get the rows to be checked
+        for (int row = current.getPosition()[1]; row > (current.getPosition()[1] - current.getShape().length); row--) {
+            boolean filled = false;
+            for (int col = 0; col < WIDTH; col++) {
+                if (grid[row][col] == 0) break;
+                // If you get to this line then it is whole row are filled. gg!
+                filled = true;
+            }
+            if (filled) {
+                line_cleared += 1;
+                for (int current_row = row; current_row < HEIGHT - 1; current_row++) {
```

Comment:
I am not sure but you seems forgot to clear the last row
also the highest row displayed is 0, while the leftmost col is still 0
I am not sure if you know